library/std/src/panic.rs

```diff
@@ -499,27 +498,26 @@ pub fn get_backtrace_style() -> Option<BacktraceStyle> {
         // to optimize away callers.
         return None;
     }
-    if let Some(style) = BacktraceStyle::from_u8(SHOULD_CAPTURE.load(Ordering::Acquire)) {
-        return Some(style);
-    }
-
-    let format = crate::env::var_os("RUST_BACKTRACE")
-        .map(|x| {
-            if &x == "0" {
-                BacktraceStyle::Off
-            } else if &x == "full" {
-                BacktraceStyle::Full
-            } else {
-                BacktraceStyle::Short
-            }
-        })
-        .unwrap_or(if crate::sys::FULL_BACKTRACE_DEFAULT {
-            BacktraceStyle::Full
-        } else {
-            BacktraceStyle::Off
-        });
-    set_backtrace_style(format);
-    Some(format)
+    BacktraceStyle::from_u8(SHOULD_CAPTURE.load(Ordering::Acquire)).map(Some).unwrap_or_else(|| {
+        let env_var = crate::env::var_os("RUST_BACKTRACE");
+        let style = env_var
+            .and_then(|var| {
+                var.to_str().map(|s| match s {
+                    "0" => BacktraceStyle::Off,
+                    "full" => BacktraceStyle::Full,
+                    _ => BacktraceStyle::Short,
+                })
```

Comment:
Note that this isn't equivalent to the equality `&x == "0"` checks that were there before. `PartialEq` for `OsStr` is usually just a byte for byte comparison, but calling `to_str` incurs the cost of validating it is UTF-8.

Unlikely to make much of a difference here, but also no reason to change it.

Reply:
thanks for accurate reviewing!
I didn't take account into the cost of validating it's UTF-8 by using `to_str`.